src/syntax/parser.rs

```diff
@@ -72,106 +72,90 @@ where
     }
 }
 
-#[derive(PartialEq, Copy, Clone)]
-enum CommentLevel {
-    Comment = 0,
-    GroupComment = 1,
-    ResourceComment = 2,
-}
-
-fn get_comment_start<I>(ps: &mut ParserStream<I>, level: &CommentLevel) -> Result<()>
-where
-    I: Iterator<Item = char>,
-{
-    let depth = *level as u8;
-    for _ in 0..(depth + 1) {
-        ps.expect_char('#')?;
-    }
-
-    if !ps.current_is('\n') {
-        ps.expect_char(' ')?;
-    }
-    Ok(())
-}
-
-fn get_comment_level<I>(ps: &mut ParserStream<I>) -> Result<CommentLevel>
-where
-    I: Iterator<Item = char>,
-{
-    let mut level = CommentLevel::Comment;
-    ps.peek();
-    if ps.current_peek_is('#') {
-        ps.peek();
-        if ps.current_peek_is('#') {
-            level = CommentLevel::ResourceComment;
-        } else {
-            level = CommentLevel::GroupComment;
-        }
-    }
-    ps.reset_peek();
-    Ok(level)
-}
-
 fn get_comment<I>(ps: &mut ParserStream<I>) -> Result<ast::Comment>
 where
     I: Iterator<Item = char>,
 {
-    let level = get_comment_level(ps)?;
-    get_comment_start(ps, &level)?;
-
+    let mut level = -1;
     let mut content = String::new();
 
     loop {
-        while let Some(ch) = ps.take_char(|x| x != '\n') {
-            content.push(ch);
+        let mut i = -1;
+        while ps.current_is('#') && ((level == -1 && i < 2) || (level != -1 && i < level)) {
+            ps.next();
+            i += 1;
         }
 
-        ps.next();
+        if level == -1 {
+            level = i;
+        }
 
-        if !ps.current_is('#') || level != get_comment_level(ps)? {
-            break;
-        } else {
-            get_comment_start(ps, &level)?;
+        if !ps.current_is('\n') {
+            ps.expect_char(' ')?;
+            while let Some(ch) = ps.take_char(|x| x != '\n') {
+                content.push(ch);
+            }
+        }
+
+        if ps.is_peek_next_line_comment(level) {
             content.push('\n');
+            ps.next();
+        } else {
+            break;
         }
     }
 
     match level {
-        CommentLevel::Comment => Ok(ast::Comment::Comment { content }),
-        CommentLevel::GroupComment => Ok(ast::Comment::GroupComment { content }),
-        CommentLevel::ResourceComment => Ok(ast::Comment::ResourceComment { content }),
+        0 => Ok(ast::Comment::Comment { content }),
+        1 => Ok(ast::Comment::GroupComment { content }),
+        2 => Ok(ast::Comment::ResourceComment { content }),
+        _ => panic!("Unknown comment level!"),
     }
 }
 
 fn get_message<I>(ps: &mut ParserStream<I>, comment: Option<ast::Comment>) -> Result<ast::Entry>
 where
     I: Iterator<Item = char>,
 {
-    let id = get_private_identifier(ps)?;
+    let id = get_entry_identifier(ps)?;
 
-    ps.skip_line_ws();
+    ps.skip_inline_ws();
 
+    // XXX Syntax 0.4 compat
     let pattern = if ps.current_is('=') {
```

Comment:
I think we said we wanted to only support the newest Fluent Syntax in `fluent-rs`. Can we use `ps.expect_char('=')` without the `if` here?

Reply:
I was hoping that this is the one valve we leave until 1.0 to give us a way out in case our opposition (mine and Pike's) to your decision gains ground?

This is a very small change that I was hoping to keep lax until 1.0 when we'd strengthen the claim.

Is that ok with you?

Reply:
No, I think we should stick to the Syntax spec as close as possible. As you say, it's a small change—it's easy to bring it back if we need to. I don't see any value in keeping the double parsing behavior around in `fluent-rs` at this point.